Documentation for `netapp-trident` template (#73)
IBM/ibm-satellite-storage: config-templates/netapp/netapp-ontap-nas/20.07/README.md

# # NetApp Trident - ONTAP NAS Driver

You can use the `netapp-ontap-nas` Satellite storage template to deploy NetApp storage drives which you can use dynamically provision and mange file volumes in your ONTAP NAS.

## Prerequisites

**Planning consideration for Infra Admin**
* Create a cluster that meets the requirements for ONTAP NAS. For more information, see the [NetApp documentation](https://netapp-trident.readthedocs.io/en/stable-v20.07/support/requirements.html). Verify that your backend ONTAP cluster is configured as a Trident backend.
   * You must have a dedicated SVM for Trident. Volumes that are created by Trident are created in this SVM.
   * You must have one or more aggregates assigned to the SVM.
   * You must have one or more dataLIFs for the SVM. Depending on the protocol used (NFS/iSCSI), at least one dataLIF is required.
   * You must have NFS services enabled on the SVM.
   * You must set up a snapshot policy on the SVM.
* Share following details with Location Admin
Review the template parameters and retrieve the values from your NetApp cluster.

**Planning consideration for Location Admin**

> * Make sure `netapp-trident` is deployed on the taregt cluster

Reviewer: ```suggestion
* Make sure that the `netapp-trident` template is deployed on the target clusters.
```

## Creating the NetApp Ontap-NAS Driver storage configuration

Create a Satellite storage configuration that uses your customized `netapp-ontap-nas` template.

**Example `sat storage config create` command**
Create a Satellite storage configuration by using the `netapp-ontap-nas` template.
```
ibmcloud sat storage config create --name 'ontapnas' --template-name 'netapp-ontap-nas' --template-version '20.07' -p 'managementLIF=10.0.0.1' -p 'dataLIF=10.0.0.2' -p 'svm=svm-nas' -p "username=admin" -p "password=<admin password>" -p "exportPolicy=nfsexport"
```

## Creating the storage assignment
Assign your `netapp-ontap-nas` storage configuration to your clusters.
**Example `sat storage assignment create` command**

```
ibmcloud sat storage assignment create --name 'nas-driver' --group <group name> --config 'ontapnas'
```

## Verifying your NetApp Ontap-NAS Driver storage configuration is assigned to your clusters

Verify that your `netapp-ontap-nas` configuration is successfully assigned to your clusters. Run the following commands to verify that the driver pods and other Kubernetes resources are deployed.

```
oc get pods 
```

```
oc get sc | grep sat
```

## Reference

> - https://netapp-trident.readthedocs.io/en/stable-v20.07/kubernetes/operations/tasks/backends/ontap/ontap-nas/index.html

Reviewer: ```suggestion
- [NetApp documentation](https://netapp-trident.readthedocs.io/en/stable-v20.07/kubernetes/operations/tasks/backends/ontap/ontap-nas/index.html)
```

> - Support: https://netapp-trident.readthedocs.io/en/stable-v20.10/support/support.html

Reviewer: ```suggestion
- [Support documentation](https://netapp-trident.readthedocs.io/en/stable-v20.10/support/support.html)
```
